### PR TITLE
Feature/add enum for status

### PR DESCRIPTION
**Pull request by perrettecl**

### CODE REVIEW

File: src/fty_srr_dto.cc [hunk 1]
```diff
@@ -28,10 +28,136 @@
...
+        std::string actionToString(Action action)
+        {
+            try
+            {
+                return actionInString.at(action);
+            }
+            catch(...)
+            {
+                return "unknown";
+            }
```
Comment (boricj) [inline src/fty_srr_dto.cc]: Don't throw an exception merely for flow control. Use `std::map::find()` and check against `std::map::end()`.

File: src/fty_srr_dto.cc [hunk 2]
```diff
@@ -28,10 +28,136 @@
...
+         * @param data
+         * @param object
+         */
+        static std::map<Action, std::string> actionInString =
+        {
+            {Action::SAVE,    "save"},
+            {Action::RESET,   "reset"},
+            {Action::RESTORE, "restore"},
+            {Action::UNKNOWN, "unknown"}
+        };
```
Comment (boricj) [inline src/fty_srr_dto.cc]: Should be `const`.

File: src/fty_srr_dto.cc [hunk 3]
```diff
@@ -28,10 +28,136 @@
...
+            object = ConfigQueryDto(stringToAction(action), passPhrase, listTemp, data);
+        }
+
+        static std::map<Status, std::string> statusInString =
+        {
+            {Status::SUCCESS,           "success"},
+            {Status::PARTIAL_SUCCESS,   "partialSuccess"},
+            {Status::FAILED,            "failed"},
+            {Status::UNKNOWN,           "unknown"}
+        };
```
Comment (boricj) [inline src/fty_srr_dto.cc]: Should be `const`.

File: src/fty_srr_dto.cc [hunk 4]
```diff
@@ -28,10 +28,136 @@
...
+        std::string statusToString(Status status)
+        {
+            try
+            {
+                return statusInString.at(status);
+            }
+            catch(...)
+            {
+                return "unknown";
+            }
```
Comment (boricj) [inline src/fty_srr_dto.cc]: Same as above, don't use exceptions just for flow control.

**Comment by MimiXav** [comment]
Like, JB, Thx for the JOB

**Pull request merged**